### ISSUE TITLE
`<xutility>`: Use addition and multiplication overflow check MSVC intrinsics like `_add_overflow_i8`, `_mul_overflow_i16`, and `_mul_full_overflow_i8`

### ISSUE THREAD

**Issue by AlexGuteniev** [microsoft/STL#4780]
Inspired by #4776 

There are Clang intrinsics used in `_Mul_overflow` and `_Add_overflow`

MSVC has its own `_mul_full_overflow_u8` and others. 
Using them would certainly result in a better codegen than the generic implementation.

Caveats:
 * Only for x86 / x64, some are only for x64
 * Not a complete set of all operations possible
 * Not `constexpr`, unlike the Clang one
 * Need to include `<intrin.h>`

These might warrant the need of another internal header for `_Mul_overflow` and `_Add_overflow`, used by `<ranges>`, `<mdspan>`, and `<numeric>` after #4776.

**Comment by AlexGuteniev** [comment]
Another caveat is maybe DevCom-10668881 -- need a good coverage

**Comment by StephanTLavavej** [comment]
Yeah, these were implemented on 2023-04-17 by internal MSVC-PR-456118 "Add overflow detection functions for addition, subtraction and multiplication on x86 and x64".

We'd eventually want to move them to `intrin0.inl.h` for the STL's use, if we want to go in this direction. We'd need to work around the known x86 codegen bug.

I'd like to understand the potential performance impact of this change before deciding whether to invest the contributor/maintainer effort in making/reviewing such a change, but if it's significant then this could be a reasonable thing to do.

**Comment by AlexGuteniev** [comment]
> We'd need to work around the known x86 codegen bug.

After thinking about possible workaround I conclude that the best would be not to use the broken variant/platform combinations. Any workaround that doesn't too much interfere the optimization might be itself optimized away.

It is is only x86 and not x64, this isn't too much impact.

**Comment by StephanTLavavej** [comment]
I believe this exceeds our capacity to handle for the foreseeable future.
